src/main/java/com/exadel/discount/entity/Discount.java

```diff
@@ -0,0 +1,40 @@
+package com.exadel.discount.entity;
+
+import lombok.*;
+import org.hibernate.annotations.GenericGenerator;
+
+import javax.persistence.*;
+import java.util.Set;
+import java.util.UUID;
+
+@Entity
+@Table(name = "discounts")
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@EqualsAndHashCode(exclude = {"tags"})
+@ToString(exclude = {"tags"})
+public class Discount {
+    @Id
+    @Column(name = "id")
+    @GeneratedValue(generator = "UUID")
+    @GenericGenerator(
+            name = "UUID",
+            strategy = "org.hibernate.id.UUIDGenerator"
```

Comment:
@Id
    @GeneratedValue(strategy = GenerationType.AUTO)
works for me nice too.

Reply:
+